File: assembla_cli/commands/status.py
from __future__ import annotations

import click

from assembla_cli.formatting import print_json, print_table


@click.group()
def status():
    """Manage ticket statuses."""


@status.command("list")
@click.option("--json", "as_json", is_flag=True, help="Output as JSON")
@click.pass_context
def status_list(ctx, as_json):
    """List all ticket statuses in the space."""
    client = ctx.obj["client"]
    space = ctx.obj["space"]
    data = client.get(f"/spaces/{space}/tickets/statuses")
    if as_json:
        print_json(data)
    else:
        print_table(
            data or [],
            columns=["id", "name", "state", "list_order"],
            headers=["ID", "Name", "State", "Order"],
        )

File: cmd/status.go
package cmd

import (
	"fmt"

	"github.com/eugene-software/assembla-cli/internal"
	"github.com/spf13/cobra"
)

var statusCmd = &cobra.Command{
	Use:   "status",
	Short: "Manage ticket statuses",
}

var statusListCmd = &cobra.Command{
	Use:   "list",
	Short: "List ticket statuses",
	RunE: func(cmd *cobra.Command, args []string) error {
		space := requireSpace()
		asJSON, _ := cmd.Flags().GetBool("json")

		data, err := Client.Get(fmt.Sprintf("/spaces/%s/tickets/statuses", space), nil)
		if err != nil {
			return err
		}

		if asJSON {
			internal.PrintJSON(data)
		} else {
			internal.PrintTable(
				internal.ToSlice(data),
				[]string{"id", "name", "state", "list_order"},
				[]string{"ID", "Name", "State", "Order"},
			)
		}
		return nil
	},
}

func init() {
	statusListCmd.Flags().Bool("json", false, "Output as JSON")

	statusCmd.AddCommand(statusListCmd)
}
